### DISCUSSION TITLE
burn tensor and ndarray

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#2251] [Q&A]
Hi,
Rust tensor and ndarray looks very similar, what's the relationship between  them ?
I feel that ndarry API is designed too complicated and hard to use. Can I replace ndarray with tensor in my daily work (such as linear algebra operation)?

**Comment by laggui** [comment]
When you use the ndarray backend, the `Tensor` struct wraps the ndarray primitive and operations are performed using the ndarray API.

You can certainly use the tensor API to perform similar operations! I guess it depends what kind of linalg ops you are using. You can see the list of operations supported on tensors here: https://burn.dev/burn-book/building-blocks/tensor.html#tensor-operations
